client/hosting-overview/index.tsx

```diff
@@ -0,0 +1,22 @@
+import page, { type Context } from '@automattic/calypso-router';
+import { makeLayout, render as clientRender } from 'calypso/controller';
+import getPrimarySiteId from 'calypso/state/selectors/get-primary-site-id';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import hostingOverview from './controller';
+
+export default function () {
+	page( '/hosting-overview', ( { store: { getState } }: Context ) => {
+		const state = getState();
+
+		const primarySiteId = getPrimarySiteId( state );
+		const selectedSite = getSelectedSite( state );
+
+		page.redirect(
+			`/hosting-overview/${
+				selectedSite ? selectedSite?.slug : getSiteSlug( state, primarySiteId )
+			}`
+		);
+	} );
+	page( '/hosting-overview/:site', hostingOverview, makeLayout, clientRender );
```

Comment:
Actually, it's here where we should perform the site switch. There should be a controller before `hostingOverview` that changes the site selector to the site given in the `:site` param.

Reply:
Addressed on https://github.com/Automattic/wp-calypso/pull/89476/commits/ecf86e58115375b3c0859fd725947a0d73c4e4ba.